src/components/QueryPicker/components/FilterQueries/FilterQueries.tsx

```diff
@@ -0,0 +1,164 @@
+import React, {
+  FC,
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+  useRef,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { transparentize } from 'polished';
+import { Dropdown } from '@keen.io/ui-core';
+import { BodyText } from '@keen.io/typography';
+import { colors } from '@keen.io/colors';
+
+import { FilterItem, SearchTags } from './components';
+import {
+  Container,
+  TagsContainer,
+  DropdownContent,
+  EmptySearch,
+  ClearFilters,
+  Filter,
+} from './FilterQueries.styles';
+
+type Props = {
+  /** Tags available */
+  tagsPool: string[];
+  /** Active tags filters */
+  tagsFilters: string[];
+  /** Cache queries filters indicator */
+  showOnlyCachedQueries: boolean;
+  /** Update cache filter event handler */
+  onUpdateCacheFilter: (isActive: boolean) => void;
+  /** Update tags filters event handler */
+  onUpdateTagsFilters: (tags: string[]) => void;
+  /** Clear filters event handler */
+  onClearFilters: () => void;
+};
+
+const FilterQueries: FC<Props> = ({
+  tagsPool,
+  tagsFilters,
+  showOnlyCachedQueries,
+  onUpdateCacheFilter,
+  onUpdateTagsFilters,
+  onClearFilters,
+}) => {
+  const { t } = useTranslation();
+  const containerRef = useRef(null);
+  const [isOpen, setOpen] = useState(false);
+  const [searchMode, setSearchMode] = useState(false);
+  const [searchPhrase, setSearchPhrase] = useState('');
+
+  const filteredTags = useMemo(() => {
+    if (searchPhrase) {
+      const phrase = searchPhrase.toLowerCase();
+      return tagsPool.filter((tag) => tag.toLowerCase().includes(phrase));
+    }
+    return tagsPool;
+  }, [searchPhrase, tagsPool]);
+
+  const outsideClick = useCallback(
+    (e) => {
+      if (
+        isOpen &&
+        containerRef.current &&
+        !containerRef.current.contains(e.target)
+      ) {
+        setOpen(false);
+        setSearchPhrase('');
+        setSearchMode(false);
+      }
+    },
+    [isOpen, containerRef]
+  );
```

Comment:
I think You can use here _useOnClickOutside_ hook from @keen.io/react-hooks. Thanks to it we will no longer need to keep such logic in component ;)